### PR TITLE
Implemented cksum

### PR DESCRIPTION
**Pull request by cospplredman**
cksum is one of the posix utilities of all time.

### CODE REVIEW

File: src/cksum/cksum
Comment (Sigmanificient) [inline src/cksum/cksum:1]: Please remove the binary file and add it to gitignore list.
Make sure to add your the folder name in `BINS` variable of the global Makefile

File: src/cksum/cksum.c [hunk 1]
```diff
@@ -0,0 +1,101 @@
+#include<stdint.h>
```
Comment (Sigmanificient) [inline src/cksum/cksum.c:1]: Can you run `clang-format`? This enforce coding style consistency across all sources

File: src/cksum/cksum.c [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+		FILE*file=fopen(argv[index],"r");
+
+		if(file==NULL)
+			return EXIT_FAILURE;
+
+		size_t octets;
+		uint32_t crc = crc32(file, &octets);
+
+		printf("%u %lu %s\n", crc, octets, argv[index]);
+	}
```
Comment (Sigmanificient) [inline src/cksum/cksum.c:100]: missing `fclose`?

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

**Comment by Sigmanificient** [comment]
The core logic is a bit cryptic (mostly bit manipulation) so I couldn't tell what happening there

**Review by Sigmanificient — APPROVED** [review]

**Comment by proh14** [comment]
hello @cospplredman. thanks for your pr. please test this before we merge it. Also, be sure to provide. if you tasted all the edge cases, mention me so I can merge it. ;)

**Review by proh14 — APPROVED** [review]

**Comment by cospplredman** [comment]
My testing methodology is as follows: run both my implementation and the gnu implementation on every file in the cksum directory.

```sh
$ ./cksum *
175369053 16296 cksum
1855786587 7 cksum.1
4088649404 2179 cksum.c
1192622583 51 Makefile
197436690 157 test.sh
$ cksum *
175369053 16296 cksum
1855786587 7 cksum.1
4088649404 2179 cksum.c
1192622583 51 Makefile
197436690 157 test.sh
```

the results match as far as i can tell.

One notable difference between this implementation and the gnu implementation is that I do not handle directories, I opt to instead just return EXIT_FAILURE. The reason I prefer to fail is that there isn't a standard C way of checking if something is a directory, so any method of doing so would break me from my (very tasteful) c99 compliance, and cksum is only defined on files anyways.

File: src/cksum/cksum.c [hunk 3]
```diff
@@ -0,0 +1,97 @@
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#define NAME "cksum (canoutils)"
+#define VERSION "0.0.1"
+#define AUTHOR "cospplredman"
+
+#include "getopt.h"
```
Comment (proh14) [inline src/cksum/cksum.c:9]: Change `getopt.h` to `cgetopt.h`

File: src/cksum/cksum.c [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+int main(int argc, char **argv) {
+  if (argc < 2) {
+    printf("%s\n", usage);
+    exit(1);
+  }
+
+  int opt;
+  while ((opt = getopt(argc, argv, "v")) != -1) {
```
Comment (proh14) [inline src/cksum/cksum.c:75]: use getopt_long instead of normal getopt. do `man getopt.3` for more info

File: src/cksum/cksum.c [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      return EXIT_SUCCESS;
+      break;
+    }
+  }
+
+  int index = optind;
+  for (; index < argc; index++) {
+    FILE *file = fopen(argv[index], "r");
+
+    if (file == NULL)
```
Comment (proh14) [inline src/cksum/cksum.c:88]: If file cannot be opened print errno. `perror("cksum")`

File: src/cksum/cksum.c [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+int main(int argc, char **argv) {
+  if (argc < 2) {
+    printf("%s\n", usage);
+    exit(1);
+  }
+
+  int opt;
+  while ((opt = getopt(argc, argv, "v")) != -1) {
+    switch (opt) {
+    case 'v':
```
Comment (proh14) [inline src/cksum/cksum.c:77]: add more flags

**Review by proh14 — CHANGES_REQUESTED** [review]

**Comment by proh14** [comment]
also pull the latest changes

**Review by proh14 — APPROVED** [review]
thx ;)
